py/DREAM/Output/DistributionFunction.py

```diff
@@ -195,17 +195,18 @@ def synchrotron(self, model='spectrum', B=3.1, wavelength=700e-9, t=None, r=None
 
         data = None
         if model == 'total':
+            data = np.zeros((len(t), len(r), self.data.shape[-2], self.data.shape[-1]))
```

Comment:
Will ``len(t)`` and ``len(r)`` work of ``t`` or ``r`` is a slice? Maybe it would be safer to do ``self.grid.t[t]`` etc?